precompiles/staking/tx.go

```diff
@@ -41,8 +43,96 @@ const (
 	RedelegateAuthz = stakingtypes.AuthorizationType_AUTHORIZATION_TYPE_REDELEGATE
 	// CancelUnbondingDelegationAuthz defines the authorization type for the staking
 	CancelUnbondingDelegationAuthz = stakingtypes.AuthorizationType_AUTHORIZATION_TYPE_CANCEL_UNBONDING_DELEGATION
+	// CreateValidatorAuthz defines the authorization type for the create validator
+	CreateValidatorAuthz = 5 // TODO: define cosmos sdk
 )
 
+// CreateValidator performs create validator.
+func (p Precompile) CreateValidator(
+	ctx sdk.Context,
+	origin common.Address,
+	contract *vm.Contract,
+	stateDB vm.StateDB,
+	method *abi.Method,
+	args []interface{},
+) ([]byte, error) {
+	msg, delegatorHexAddr, err := NewMsgCreateValidator(args, p.stakingKeeper.BondDenom(ctx))
+	if err != nil {
+		return nil, err
+	}
+
+	p.Logger(ctx).Debug(
+		"tx called",
+		"method", method.Name,
+		"args", fmt.Sprintf(
+			"{ min_self_delegation: %s, delegator_address: %s, validator_address: %s, pubkey: %s, value: %s }",
+			msg.MinSelfDelegation.String(),
+			delegatorHexAddr,
+			msg.ValidatorAddress,
+			msg.Pubkey.String(),
+			msg.Value.Amount.String(),
+		),
+	)
+
+	var (
+		// stakeAuthz is the authorization grant for the caller and the delegator address
+		stakeAuthz *stakingtypes.StakeAuthorization
+		// expiration is the expiration time of the authorization grant
+		expiration *time.Time
```

Comment:
`StakeAuthorization` won't apply here, we either leave this without the need for an `authz` or we create a custom one, will discuss with the team.

Reply:
To be honest, I didn’t read these logics carefully. They all copied the **func (p Precompile) Delegate** function below. I will fix other problems based on your suggestions first.

Reply:
I've changed it to: only allow users to directly call the StakingI contract's address which is 0x0000000000000000000000000000000000000800 to create a new validator